src/actions/channels.ts

```diff
@@ -724,6 +724,36 @@ export function deleteChannel(channelId: string): ActionFunc {
     };
 }
 
+export function undeleteChannel(channelId: string): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let state = getState();
+        const viewArchivedChannels = state.entities.general.config.ExperimentalViewArchivedChannels === 'true';
+
+        try {
+            await Client4.undeleteChannel(channelId);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        state = getState();
+        const {currentChannelId} = state.entities.channels;
+        if (channelId === currentChannelId && !viewArchivedChannels) {
```

Comment:
I am still trying to understand this block of code and what dispatch(SELECT_CHANNEL)  does.